internal/helm/chart/builder_local.go

```diff
@@ -189,14 +189,18 @@ func (b *localChartBuilder) Build(ctx context.Context, ref Reference, p string,
 // mergeFileValues merges the given value file paths into a single "values.yaml" map.
 // The provided (relative) paths may not traverse outside baseDir. It returns the merge
 // result, or an error.
-func mergeFileValues(baseDir string, paths []string) (map[string]interface{}, error) {
+func mergeFileValues(baseDir string, paths []string, ignoreMissing bool) (map[string]interface{}, error) {
 	mergedValues := make(map[string]interface{})
 	for _, p := range paths {
 		secureP, err := securejoin.SecureJoin(baseDir, p)
 		if err != nil {
 			return nil, err
 		}
-		if f, err := os.Stat(secureP); err != nil || !f.Mode().IsRegular() {
+		f, err := os.Stat(secureP)
```

Comment:
maybe something like

```go
		if err != nil {
			if ignoreMissing && os.IsNotExist(err) {
			continue
			}
			return nil, fmt.Errorf("no values file found at path '%s' (reference '%s')",
			strings.TrimPrefix(secureP, baseDir), p)
		}
		if !f.Mode().IsRegular() {
			return nil, fmt.Errorf("no values file found at path '%s' (reference '%s')",
				strings.TrimPrefix(secureP, baseDir), p)
		}
```

this avoid checking the error twice.

Reply:
I was favouring the double `err != nil` which should be super-lightweight over the repeated complex `return nil, fmt.Errorf(…)`, but either works. You favour repeating the `return …`?

Reply:
How do you feel about this variant?

```go
		switch {
		case err != nil:
			if ignoreMissing && os.IsNotExist(err) {
				continue
			}
			fallthrough
		case !f.Mode().IsRegular():
			return nil, fmt.Errorf("no values file found at path '%s' (reference '%s')",
				strings.TrimPrefix(secureP, baseDir), p)
		}
```